src/Core/Crawl/iworld.cpp

```diff
@@ -61,10 +63,38 @@ bool IWorld::start() {
 
 
     worldChanged(*_worldRules->begin());
-
     return true;
 }
 
+QObject *IWorld::player() const {
+    return _player;
+}
+
+void IWorld::setPlayer(QObject *newPlayer) {
+    if (_player == newPlayer)
+        return;
+
+    auto newPlayerObject = dynamic_cast<IPlayer*>(newPlayer);
+    if (!newPlayerObject) {
+        QuasarAppUtils::Params::log("Failed to set player object. The input object is not player.",
+                                    QuasarAppUtils::Error);
+        return;
+    }
+
+    if (_player) {
+        removeIAtomicItem(_player->guiId());
+    }
+
+    _player = newPlayerObject;
+    addAtomicItem(_player);
+
+    emit playerChanged();
+}
+
+IWorldItem *IWorld::generate(const QString &objectType) const {
+    return _registeredTypes.value(objectType, [](){return nullptr;}).operator()();
```

Comment:
What it is?

Reply:
This is default generation objects function. 
the **_registeredTypes** are objects that contains lambada functions with create needed objects.
the **[](){return nullptr;}** are default lambda function that will be return when  the objectType not supported/
the **operator()()** are coll the returned lambda function

Reply:
see the [registerObject](https://github.com/QuasarApp/Crawl/pull/59/files#diff-b02ea4b54bc77c555860ecb5f905c748f0908c11271b131074ee2c424d1e52cdR316) function for get more information.